src/com/example/mutations.cljs

```diff
@@ -17,3 +19,26 @@
   (action [{:keys [state] :as env}]
           (github-signout)
           (swap! state assoc :root/user {:user/authenticated? false})))
+
+(def readiness-ident [:component/id :readiness])
+
+(defn readiness-class [] (comp/registry-key->class :com.example.ui/ReadinessForm))
+
+(defn clear-readiness-form*
+  "Mutation helper: Updates state map with a cleared readiness form that is configured for form state support."
+  [state-map]
+  (-> state-map
+      (assoc-in readiness-ident
+                {:readiness/trained ""})
+      (fs/add-form-config* (readiness-class) readiness-ident)))
+
+(defmutation clear-readiness-form [_]
+  (action [{:keys [state]}]
+          (swap! state clear-readiness-form*)))
+
+;; implement this when we want redo
+#_(defmutation use-quiz-form [{:quiz/keys [id]}]
```

Comment:
in fulcro, you send in a mutation to set up initial state for a form. then you can modify the values in your db with the guarantee that you can revert back to the original version.

Theres also a bunch of form validation stuff I havent looked into